straxbra/plugins.py

```diff
@@ -368,58 +372,80 @@ def compute(self, peaks):
                  default=10)
 )
 class PeakPositions(strax.Plugin):
-    """
-    I actually wrote this
-    Version 0.0.1: weighted sum
-    Version 0.0.2: hopefully LRF or better
-    """
+    '''
+    Position Reconstruction for XeBRA with LRFs
+
+    Status: July 2019
+
+    Position reconstruction for XeBRA following the position reconstruction algorithm of Mercury
+    (employed in the LUX experiment, originally developed for the ZEPLIN-III dark matter experiment)
+    with light response functions (see https://arxiv.org/abs/1710.02752v2 and
+    https://arxiv.org/abs/1112.1481 ).
+    '''
     __version__ = "0.0.1"
     dtype = [('x', np.float32,
-              'Reconstructed S2 X position (cm), uncorrected'),
+              'Reconstructed S2 X position (mm), uncorrected'),
              ('y', np.float32,
-              'Reconstructed S2 Y position (cm), uncorrected')]
+              'Reconstructed S2 Y position (mm), uncorrected')]
     depends_on = ('peaks',)
 
     parallel = False
 
-    pmt_locations = np.zeros((7,2))  # top only
-
     daq_to_location_map = np.arange(7)  # s2-top-channel to pmt-location
 
-    def setup(self):
-
-        self.pmt_mask = self.config['to_pe'][:self.config['top_pmts']] > 0
-        pmt_angles = np.linspace(0, 2*np.pi, 6, endpoint=False)
-        angle_offset = -2*np.pi/3  # position of PMT1
-        pmt_angles = pmt_offset - pmt_angles
-        pmt_radius = 30  # mm
-        self.pmt_positions = np.column((np.cos(pmt_angles),
-                                        np.sin(pmt_angles)))* pmt_radius
-        self.pmt_positions = np.append(self.pmt_positions, [[0,0]])
+    ## Function to return non-negative value corresponding to (radial position - radius TPC) if inside TPC,
+    ## used for constraints
+    @staticmethod
+    def insidevolume(inputs):
+        p_TPC_radius = 35
+        return (p_TPC_radius - np.sqrt(inputs[0]**2 + inputs[1]**2))
 
+    ## Radial LRF model
+    ## from: Position Reconstruction in a Dual Phase Xenon Scintillation Detector (https://arxiv.org/abs/1112.1481)
+    @staticmethod
+    def eta(r, A, r0, a, b, alpha):
+        return A * exp( - a * (r/r0) / (1 + (r/r0) ** (1 - alpha)) - b / (1 + (r/r0) ** (- alpha)))
+
+    ## Position dependent LRF values individual PMTs from model
+    def LRF_PMTs(x, y, self):
```

Comment:
`self` must be the first argument

Reply:
Thanks for the hints, they're taken to heart in the updated version.